index.js

```diff
@@ -610,6 +615,13 @@ class Batch {
 
         if (c === 0) {
           if (!this.overwrite) return this._unlockMaybe()
+
+          if (onlyIfChanged) {
+            const block = await this.getBlock(node.keys[mid].seq)
+            const same = (Buffer.compare(block.value, value) === 0)
+            if (same) try { return block.final() } finally { this._unlockMaybe() }
```

Comment:
`if (same) return this._unlockMaybe()` is fine

Reply:
that's fine, we'll tweak after or fix in the other PR :)

Reply:
sounds good!